packages/hyperion-autologging/src/ALTriggerFlowlet.ts

```diff
@@ -172,6 +145,55 @@ export function init(options: InitOptions) {
     }
   });
 
+  // Temporarily disable this until we can re-enable react-flowlet logic bellow
+  if (options.enablePerSurfaceTracking) {
+    const ALSurfaceContextDataMap = new Map<ALSurfaceContextFilledValue['surface'], ALSurfaceContextFilledValue['flowlet']>();
+
+    const interceptedSetter = interceptFunction(setActiveTriggerFlowlet);
+    setActiveTriggerFlowlet = interceptedSetter.interceptor;
+    interceptedSetter.onArgsObserverAdd((triggerFlowlet, surface) => {
+      if (surface && triggerFlowlet) {
+        // The following may not happen until react interception and flowlets actually are enabled.
+        const surfaceFlowlet = ALSurfaceContextDataMap.get(surface);
+        if (surfaceFlowlet) {
+          surfaceFlowlet.data.triggerFlowlet = triggerFlowlet;
+        }
+      }
+    });
+
+    // Track surface flowlet roots
+    channel.addListener('al_surface_mount', event => {
+      const { surface, flowlet } = event;
+      ALSurfaceContextDataMap.set(surface, flowlet);
+      let rootFlowlet = flowlet;
+      while (rootFlowlet.parent) {
+        rootFlowlet = rootFlowlet.parent;
+      }
+      activeRootFlowlets.add(rootFlowlet);
+    });
+    channel.addListener('al_surface_unmount', event => {
+      ALSurfaceContextDataMap.delete(event.surface);
+      activeRootFlowlets.delete(event.flowlet);
+    });
+  }
+
+  if (options.enableFlowletConstructorTracking) {
+    //@ts-expect-error
+    flowletManager.root.data.isRootet = true;
```

Comment:
Is this field supposed to be `isRooted`? Could you help explain a bit about what this is tracking?

Reply:
Yes, thanks for catching the typo. I added the full description as comment in the code. Adding it here for reference:
```
    /**
     * To know if root of a flowlet is added to list, we would need
     * to walk up the .parent chain to find the root.
     * Since we know most of the flowlets will be created from the
     * FlowletManager.root, we use a the following trick:
     * We add a known field the root's data and check chat in children's
     * data (data fields inherit from each other).
     * If we don't find the field, we have two options:
     * 1- Now walk up the .parent chain (slow path)
     * 2- Consider this case error and fix it in the code.
     * For now, we choose 2 since there is no real legit case for a
     * 'rootless' flowlet!
     */
```